rasa/core/brokers/pika.py

```diff
@@ -165,9 +165,18 @@ async def _connect(self) -> aio_pika.abc.AbstractRobustConnection:
         # `password`.
         if self.host.startswith("amqp"):
 
-            parsed_host = urlparse(self.host)
-            amqp_user = f"{self.username}:{self.password}"
-            url = f"{parsed_host.scheme}://{amqp_user}@{parsed_host.netloc}:{self.port}"
+            if self.username in self.host and self.password in self.host:
```

Comment:
Suggestion: you can extract this into separate component called `PikaBrokerConfigBuilder` which after initialization can provide amqp_user, parsed_host and url properties.
It can also be easy to write detailed unit tests for it.

Reply:
Great idea, however I've opted for extracting this into a separate method instead of the OOP route. Let me know what you think.